sklearn/tree/tests/test_tree.py

```diff
@@ -472,64 +472,76 @@ def test_max_features():
 
         # use values of max_features that are invalid
         est = TreeEstimator(max_features=10)
-        assert_raises(ValueError, est.fit, X, y)
+        with pytest.raises(ValueError):
```

Comment:
We should open an issue to refactor theses tests as well to use parametrization.
It might be easier to read. But let's keep it for another PR.

Reply:
Yes, that is a good point. We should do that